### ISSUE TITLE
Use of base64_encode() for SVG icons while registering custom post types

### ISSUE THREAD

**Issue by boonebgorges** [WordPress/plugin-check-legacy#53]
Previously: #24 

Hi folks - Thanks for the valuable work on this plugin.

I noticed that the following usage is flagged as forbidden because of the base64_encode() check:

```php
register_post_type(
  'my_post_type',
  array(
     // ...
     'menu_icon' => 'data:image/svg+xml;base64,' . base64_encode( '<svg xmlns="http://www.w3.org/2000/svg" height="24" width="24"><path fill="black" d="M2 9V7H7V9ZM2 14V12H7V14ZM20.6 19 16.75 15.15Q16.15 15.575 15.438 15.787Q14.725 16 14 16Q11.925 16 10.463 14.537Q9 13.075 9 11Q9 8.925 10.463 7.462Q11.925 6 14 6Q16.075 6 17.538 7.462Q19 8.925 19 11Q19 11.725 18.788 12.438Q18.575 13.15 18.15 13.75L22 17.6ZM14 14Q15.25 14 16.125 13.125Q17 12.25 17 11Q17 9.75 16.125 8.875Q15.25 8 14 8Q12.75 8 11.875 8.875Q11 9.75 11 11Q11 12.25 11.875 13.125Q12.75 14 14 14ZM2 19V17H12V19Z"/></svg>' ),
    //
  )
);
```

My reading of https://developer.wordpress.org/reference/functions/register_post_type/#menu_icon is that this is a recommended way to supply a custom icon when registering a post type. 

I'm unsure how to make PHPCS ignore the error in this specific type of case (or even whether it's possible to do so).

For my purposes I can move the icon into an image file, but I wanted to flag this in case the team was interested. Feel free to close if it's not germane to your conversation. Thanks again for your work!

**Comment by rawrly** [comment]
Hi Boone,

It is possible to tell phpCS to ignore a line of code by appending a comment to the line telling phpCS to ignore that line. 

It uses a format of `// phpcs:ignore <rule to ignore> -- further comments`

Using your example, it would look something like this:

```
  array(
     // ...
     'menu_icon' => 'data:image/svg+xml;base64,' . base64_encode( '<svg xmlns="http://www.w3.org/2000/svg" height="24" width="24"><path fill="black" d="M2 9V7H7V9ZM2 14V12H7V14ZM20.6 19 16.75 15.15Q16.15 15.575 15.438 15.787Q14.725 16 14 16Q11.925 16 10.463 14.537Q9 13.075 9 11Q9 8.925 10.463 7.462Q11.925 6 14 6Q16.075 6 17.538 7.462Q19 8.925 19 11Q19 11.725 18.788 12.438Q18.575 13.15 18.15 13.75L22 17.6ZM14 14Q15.25 14 16.125 13.125Q17 12.25 17 11Q17 9.75 16.125 8.875Q15.25 8 14 8Q12.75 8 11.875 8.875Q11 9.75 11 11Q11 12.25 11.875 13.125Q12.75 14 14 14ZM2 19V17H12V19Z"/></svg>' ), // phpcs:ignore Generic.PHP.ForbiddenFunctions -- base64 encoding is recommended for custom icons
     // ...
```

Source and further reading can be found at: https://docs.wpvip.com/how-tos/customize-phpcs/

**Comment by boonebgorges** [comment]
Thanks for the quick response. I'm aware that I can ignore in my own code, but I wasn't sure whether this kind of `ignore` would be a recommended practice for users of Plugin Check (as well as the team's internal tools), since you could just `ignore` anything sneaky you're trying to put into the plugin.

**Comment by rawrly** [comment]
The purpose is more for the safety and security of plugins code, catching accidental security bugs added to code and providing a tool to empower developers to write secure code.

If a developer tells phpCS to ignore their insecure, obfuscated, or guideline violating code. That will be quickly spotted (by reviewing the lines they tell phpCS to ignore) and would raise valid and legitimate concerns about the developer's intent.

**Comment by boonebgorges** [comment]
Sounds good. Thanks for providing clarity on the goals of the plugin. I'll close this ticket.
